projects/web-wallet/rpc-protocol.md

```diff
@@ -0,0 +1,432 @@
+# 1. Introduction
+
+This document outlines the remote procedure call (RPC) methods for a decentralized application (dApp) to interact with Hathor wallets.
+
+The design is currently a minimum viable product (MVP), addressing only known needs from our community. It should be noted that this API design may change as the platform evolves and additional needs arise.
+
+The idea here is to partially replicate the headless wallet API as it is already being used by many partners and has been validated to work on their use cases
+
+These methods are going to be implemented in the wallet, and exposed to dApps using wallet-connect, the design of the integration can be read [here](https://github.com/HathorNetwork/rfcs/blob/master/projects/web-wallet/wallet-connect/0001-design.md) and the wallet-connect design docs can be read [here](https://docs.walletconnect.com/web3wallet/about).
+
+# 2. Reference-level explanation
+
+## 2.1 RPC Methods
+
+### 2.1.1 `htr_sendTx`
+
+The `htr_sendTx` method creates a new transaction based on parameters. It will select the `utxos` automatically, mine and send the transaction
```

Comment:
Should we add an optional `push_tx: bool` to allow operations where the transaction will not be pushed by the wallet?

Reply:
We can, added it in b682c3818f1e2e5f884053518abdbde8668969aa

I also added a new RPC method to push a transaction hex, what do you think?

Reply:
It's fine. This new method will be called by the wallet, right?

Reply:
Yes, the dApp will send a RPC to the wallet and the wallet will execute it